codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamFrameToHttpObjectCodec.java

```diff
@@ -120,6 +133,17 @@ private void encodeLastContent(LastHttpContent last, List<Object> out) {
 
     @Override
     protected void encode(ChannelHandlerContext ctx, HttpObject obj, List<Object> out) throws Exception {
+        // 100-continue is typically a FullHttpResponse, but the decoded
+        // Http2HeadersFrame should not be marked as endStream=true
+        if (obj instanceof FullHttpResponse) {
```

Comment:
in the event of separate sequential HttpResponse (status=100) and LastHttpContent objects being written, the following objects would be encoded:

```
1. Http2HeadersFrame(status:100, endStream=false)
2. Http2HeadersFrame(endStream=true)
```
unless we maintain some kind of state in this handler to swallow the LastHttpContent, this will result in the stream to close immediately after 100-continue is written. i was hoping that we don't have to maintain that state, as devs could simply write a FullHttpResponse instead...

Reply:
@lionelli1 I think we should just mention in the java docs that this is needed and throw an exception if the dev tries to not use a FullHttpResponse here. Can you do this and also add a unit test ?

Reply:
good point. changed and added unit test to verify behavior.

Reply:
<3